drivers/ospi/src/ospi.c

```diff
@@ -226,21 +236,24 @@ void ospi_send(OSPI_Type *ospi, ospi_transfer_t *transfer)
 }
 
 /**
- * \fn          void ospi_receive(OSPI_Type *ospi, ospi_transfer_t *transfer)
+ * \fn          void ospi_receive(struct ospi_regs *ospi,
+ *                                struct ospi_transfer *transfer)
  * \brief       Prepare the OSPI instance for reception
  * \param[in]   ospi       Pointer to the OSPI register map
  * \param[in]   transfer   Transfer parameters
  * \return      none
  */
-void ospi_receive(OSPI_Type *ospi, ospi_transfer_t *transfer)
+void ospi_receive(struct ospi_regs *ospi, struct ospi_transfer *transfer)
 {
 	uint32_t val;
 
 	ospi_disable(ospi);
 
 	val = ospi->OSPI_CTRLR0;
-	val &= ~(SPI_CTRLR0_SPI_FRF_MASK | (SPI_CTRLR0_TMOD_MASK | SPI_CTRLR0_SSTE_MASK));
-	val |= ((transfer->spi_frf << SPI_CTRLR0_SPI_FRF) | SPI_CTRLR0_TMOD_RECEIVE_ONLY);
+	val &= ~(SPI_CTRLR0_SPI_FRF_MASK
```

Comment:
This code is at multiple time so it could be local function for update ospi->OSPI_CTRLR0.  It will help code maintence in future also

Reply:
I re check and only changed part is SPI_CTRLR0_TMOD_SEND_ONLY or SPI_CTRLR0_TMOD_RECEIVE_ONLY so that could be a parameter with ospi->OSPI_CTRLR0 to uint32_t
for example:
`ospi_control0_set(ospi->OSPI_CTRLR0, SPI_CTRLR0_TMOD_RECEIVE_ONLY)`